### PR TITLE
Remchain to ERC20 Rem swap function

### PR DESCRIPTION
**Pull request by Art17**
Allows an owner to submit and confirm a swap transaction from Remchain to ERC20 REM.
Parameters:
 - destination - Transaction target address.
 - value - Transaction ether value.
 - nonce - Transaction counter in Remchain.
 - data - Transaction data payload.

Returns transaction ID.

### CODE REVIEW

File: contracts/MultiSigWallet.sol [hunk 1]
```diff
@@ -27,11 +27,12 @@ contract MultiSigWallet {
      *  Storage
      */
     mapping (uint => Transaction) public transactions;
+    mapping (bytes32 => uint) public swapTransactions;
```
Comment (roman-tik) [inline contracts/MultiSigWallet.sol]: why do we need this map? maybe we just include bytes32 variable to the Transaction?

File: contracts/MultiSigWallet.sol [hunk 2]
```diff
@@ -194,6 +195,25 @@ contract MultiSigWallet {
         confirmTransaction(transactionId);
```
Comment (roman-tik) [inline contracts/MultiSigWallet.sol:195]: I think we do not need submitTransaction any longer

File: contracts/MultiSigWallet.sol [hunk 3]
```diff
@@ -194,6 +195,25 @@ contract MultiSigWallet {
...
     }
 
+    /// @dev Allows an owner to submit and confirm a swap transaction from Remchain to ERC20 REM.
+    /// @param destination Transaction target address.
+    /// @param value Transaction ether value.
+    /// @param nonce Transaction counter in Remchain.
+    /// @param data Transaction data payload.
+    /// @return Returns transaction ID.
+    function submitAndConfirmSwapTransaction(address destination, uint value, uint nonce, bytes data)
```
Comment (roman-tik) [inline contracts/MultiSigWallet.sol]: maybe just SumbitSwapTransaction?

Comment (Art17) [reply contracts/MultiSigWallet.sol]: ProcessSwapTransaction

File: contracts/MultiSigWallet.sol (hunk 2 above)
Comment (Art17) [reply contracts/MultiSigWallet.sol:195]: Decided to support multisig interface.

File: contracts/MultiSigWallet.sol (hunk 1 above)
Comment (Art17) [reply contracts/MultiSigWallet.sol]: Done.

**Pull request merged**